### ISSUE TITLE
percent_diff() returns the absolute value 

### ISSUE THREAD

**Issue by tinohean** [grafana/grafana#10129]
VERSION:Grafana v4.6.1
DATASOURCE: mysql

need alert when value declined 50% comared with 6 mins earlier,
and I used percent_diff() which returns absolute value only , so I added like this

```
    WHEN diff() OF query(A,16m,now-10m) IS BELOW 0
    AND  percent_diff() OF query(A,16m,now-10m) IS ABOVE 50
```

the problem is I got too many metrics in query A.
got wrong alert when metrics_1 declined and metrics_2 rosed by 60% .

**Comment by bergquist** [comment]
This could solve by introducing two new reducers. `percent_diff_increase` and `percent_diff_decrease`

**Comment by zarbis** [comment]
@bergquist I think current `diff` and `percent_diff` are misleading since they compare `latest` and `latest-1` points no matter what time interval you set in query. In other words: they compare 2 closest possible most recent points, when in my opinion they should compare 2 furthest possible points within specified time interval.

**Comment by tcgjensen** [comment]
@bergquist is this something that will be prioritized? I have tested it and it seems @zarbis is correct, it uses the latest and latest-1 datapoints nomatter which interval you choose. Also it is rather inconvenient that you are not able to indicate whether you are interested in a percentage increase or decrease. Now you get alerts nomatter which one it is.

**Comment by mykllterry** [comment]
Currently using  Grafana v5.0.0 (commit: af6e283) 
I've run into the issue described above.
I did the same as the initial user and made one condition to check for a negative change  and then check for the percent_diff. 
It worked and then would immediately clear the next check interval. That seems to match with the above description about which points are being checked.

I'm wondering if there is any work being done to address the question in this thread?

**Comment by marefr** [comment]
@mykllterry please try Grafana v5.3.0 which include fix of #11563. Details of fix [here](https://github.com/grafana/grafana/pull/12515)

**Comment by martinbruntsch** [comment]
Hi @marefr , 
I would like to take this issue and create a PR for it. 
But before starting, I want to align .. please give me some feedback before I go ahead

Firstly, I will fix the issue https://github.com/grafana/grafana/issues/16270 and 
* rename the current reducer "diff" and "percent_diff" to "diff_abs" and "percent_diff_abs"

Secondly, I will fix teh issue https://github.com/grafana/grafana/issues/10129 and 
* provice a new reducer "diff" and "percent_diff" which is comparable to "diff_abs" and "percent_diff_abs" but return the signed differences

**Comment by danieldistler** [comment]
Possible workaround:
Add the difference transformation to the select clause of the query and use the sum reducer in the query condition. The sum of all differences is the same as the difference of the first and the last value.

For the percent case: Add a second condition and check if the difference is below 0.

**Comment by gfoligna** [comment]
Still getting this bug on v6.6.0.

**Comment by bergquist** [comment]
This have been fixed in https://github.com/grafana/grafana/pull/21338 and is included in 6.7.
we released 6.7 beta last week.
